akka-cluster/src/main/scala/akka/cluster/Cluster.scala

```diff
@@ -333,6 +337,11 @@ class Cluster(val system: ExtendedActorSystem) extends Extension {
   def joinSeedNodes(seedNodes: java.util.List[Address]): Unit =
     joinSeedNodes(Util.immutableSeq(seedNodes))
 
+  private def checkHostCharacters(address: Address): Unit =
+    require(
+      !address.hasInvalidHostCharacters,
+      s"Joining the cluster using invalid host characters '${address.host}' in the Address is not allowed.")
```

Comment:
Not saying I like a function that has a Unit signature which can throw.... but here we want to fail fast vs return something to handle.